performance-storage-service/pss_project/api/github_integration/NoisePageRepoClient.py

```diff
@@ -0,0 +1,108 @@
+import time
+import requests
+from github3 import GitHub
+from jwt import JWT, jwk_from_pem
+
+from pss_project.api.constants import REPO_OWNER, REPO_NAME, GITHUB_BASE_URL
+
+
+class NoisePageRepoClient():
+    def __init__(self, private_key, app_id):
+        """ Connect to github and create a Github client and a client specific
+        to the Github app installation"""
+        self.private_key = private_key
+        self.owner = REPO_OWNER
+        self.repo = REPO_NAME
+
+        self.git_client = GitHub()
+        self.git_client.login_as_app(private_key_pem=str.encode(private_key), app_id=app_id)
+        self.noisepage_repo_client = self.git_client.app_installation_for_repository(self.owner, self.repo)
```

Comment:
I google and found that there is no GitHub python library support check-run now. So we only use github3 for authentication and use `requests` to directly call Github restful API. Did I understand correctly?

Reply:
Yup. We also use it to check the app id against incoming requests, but that is all for now. Maybe it provides more help in the future but who knows.